src/components/RequireToken.jsx

```diff
@@ -0,0 +1,12 @@
+import { useNavigate } from 'react-router-dom';
+import { useEffect } from 'react';
+
+const RequireToken = ({ children }) => {
+  const navigate = useNavigate();
+  useEffect(() => {
+    if (!localStorage.getItem('token')) navigate('/');
+  }, [navigate]);
+  return children;
+};
+
+export default RequireToken;
```

Comment:
This looks great, Rafa! I'm curious because I haven't done this before, what does putting `navigate` as the dependency is the `useEffect()` do? Does the value of `navigate` change? I'd love to learn more :)

Reply:
It's just one of those React idiosyncrasies; to my understanding, any variable used within a useEffect hook has to exist within the dependency array, whether it changes or not (unless you use a useRef hook, but I'm really shaky on that). The useNavigate hook apparently returns the function that you then use to navigate, hence why it has to be assigned to a variable prior.

Reply:
I am still getting comfortable with props.children. You are teaching me so much, Rafa!